src/vs/editor/common/model/textModel.ts

```diff
@@ -1461,6 +1461,7 @@ export class TextModel extends Disposable implements model.ITextModel {
 				const currentEditStartLineNumber = newLineCount - lineCount - changeLineCountDelta + startLineNumber;
 				const firstEditLineNumber = currentEditStartLineNumber;
 				const lastEditLineNumber = currentEditStartLineNumber + editingLinesCnt;
+				const lastInsertedLineNumber = currentEditStartLineNumber + insertingLinesCnt - editingLinesCnt;
```

Comment:
I don't think this is correct in the case of e.g. selecting 10 lines and pasting 3 lines. I think this needs to be:
```suggestion
				const lastInsertedLineNumber = currentEditStartLineNumber + insertingLinesCnt - deletingLinesCnt;
```

Reply:
It cannot be `- deletingLinesCnt`.

```ts
const startLineNumber = change.range.startLineNumber;
const endLineNumber = change.range.endLineNumber;

const deletingLinesCnt = endLineNumber - startLineNumber;
```

Deleting lines count would be 9 when replacing 10 lines.

`insertingLinesCnt` is 2 when replacing 10 lines with 3 lines.

I don't fully understand the logic of `currentEditStartLineNumber`, but in my debug session it is 3 (probably because the edit starts with line 3).

![image](https://user-images.githubusercontent.com/2931520/123558101-e611f400-d794-11eb-8e26-31e21ec63257.png)

With your suggestion, `lastInsertedLineNumber` would evaluate to `3 + 2 - 9 = -4`. This cannot be true.

I think the line must be simply this:
```
const lastInsertedLineNumber = currentEditStartLineNumber + insertingLinesCnt;
```

This is independent of how many lines got replaced.

Reply:
> I think the line must be simply this:
```
const lastInsertedLineNumber = currentEditStartLineNumber + insertingLinesCnt;
```
:+1: